### ISSUE TITLE
Origin and cache should be able to control downtime

### ISSUE THREAD

**Issue by bbockelm** [PelicanPlatform/pelican#1251]
The director administrator can force a cache or origin into downtime centrally.

However, I don't want this power to only exist in the hands of the director administrator.  The origin/cache administrator should have a web UI (and a corresponding CLI) where they can put their own service in and out of downtime.

**Comment by CannonLock** [comment]
@howard we are going to have to split this ticket into a origin/cache api endpoint and the appropriate UI. Typically Haoming would build the api and I would build on top of it. Does that sound good to you?

**Comment by howard** [comment]
@CannonLock I'm not sufficiently familiar with this project to give any advice.

**Comment by h2zh** [comment]
Sounds good, @CannonLock Could you send me a few links to your previous collaborations so that I can learn more?

**Comment by h2zh** [comment]
This ticket is 80% done. However, it is blocked by an existing problem: The Director serverAds is a dictionary with key-value pairs as {serverURL:serverAd}. The format of serverURL looks like `https://r20abdfaecf7f:8443`. The namespaces registered in the registry sqlite db also follow this pattern. However, they are storing different PORTS - the director stores xrootd data port (e.g. 8843), while the registry saves web port (e.g. 8847)

When an origin wants to put itself into downtime, it sends a request to the director. Director first checks if the sender's server url is stored in director's cached serverAds. If so, it will retrieve this advertisement and carry the server namespace (e.g. `/origins/test-server-name`) to ask the registry if this namespace is registered. The inconsistency of PORTS break this workflow.

I talked to Justin about this. He mentioned there is another ticket https://github.com/PelicanPlatform/pelican/issues/1241 that could solve this blocker. So I will just put this ticket on hold. When I completed that ticket, I'll go back to this one.

**Comment by h2zh** [comment]
I should've tagged you here for the update! @CannonLock
